### PR TITLE
[KernelGen][Nvidia] Add batch_norm_backward operator with Triton kernel

### PR DESCRIPTION
**Pull request by bwbwzzz**
## Summary
Adds a Triton kernel for `batch_norm_backward`. This is a cuDNN-compatible wrapper that converts the cuDNN interface (save_var, update) to the native interface (save_invstd, train), enabling the Triton-based implementation to serve both `native_batch_norm_backward` and `batch_norm_backward` aten ops.

## Testing
- Parametrized tests over 5 different shapes and float16/float32 dtypes (20 cases)
- Validated against PyTorch reference on device side
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_batch_norm_backward.py --level core` (NVIDIA H20)

### float16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| [4, 16, 64, 4] | 0.008 | 0.012 | 0.627 |
| [16, 16, 8, 48] | 0.017 | 0.013 | 1.276 |
| [16, 16, 8, 88] | 0.025 | 0.014 | 1.755 |
| [16, 16, 128] | 0.009 | 0.012 | 0.789 |
| [20, 6, 65536] | 1.852 | 0.285 | 6.505 |

### float32
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| [4, 16, 64, 4] | 0.008 | 0.012 | 0.634 |
| [16, 16, 8, 48] | 0.013 | 0.013 | 1.017 |
| [16, 16, 8, 88] | 0.018 | 0.016 | 1.134 |
| [16, 16, 128] | 0.009 | 0.011 | 0.773 |
| [20, 6, 65536] | 1.777 | 0.412 | 4.317 |

### bfloat16
| Size Detail | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|---|---|---|---|
| [4, 16, 64, 4] | 0.008 | 0.013 | 0.613 |
| [16, 16, 8, 48] | 0.017 | 0.066 | 0.260 |
| [16, 16, 8, 88] | 0.025 | 0.014 | 1.853 |
| [16, 16, 128] | 0.009 | 0.012 | 0.822 |
| [20, 6, 65536] | 1.886 | 0.311 | 6.068 |

**Arithmetic Mean Speedup: 1.90x**

## Multi-backend Testing
| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---|---|---|---|
| Nvidia (H20) | PASS | PASS (15 cases, --level core) | 1.90x | Primary |
| Tianshu | PASS | PASS | 4.30x | — |
| Muxi | FAIL | — | — | Accuracy N/A |
| Ascend | FAIL | — | — | Accuracy N/A |
| Hygon | PASS | PASS | 2.24x | — |

## Files Changed
- `src/flag_gems/ops/batch_norm_backward.py`: cuDNN wrapper implementation
- `tests/test_batch_norm_backward.py`: Accuracy test
- `benchmark/test_batch_norm_backward.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Update labels and for fields

### CODE REVIEW

**Review by Dongxu-H — CHANGES_REQUESTED** [review]
## Code Review Summary

### Overview
This PR adds Triton kernel implementation for `batch_norm_backward` operator with a cuDNN-compatible interface wrapper. The implementation achieves an average **1.90x speedup** on Nvidia H20.

### 🔴 Critical Issues (Must Fix Before Merge)

#### 1. CI Test Failure 🚨
```
python-op: FAIL (1m22s)
```
The main CI test suite is failing. This must be investigated and fixed before the PR can be merged.

#### 2. Muxi Backend Test Failure 🚨
```
Muxi: FAIL - Accuracy test failed
```
This indicates compatibility issues with the Muxi backend.

#### 3. Ascend Backend Test Failure 🚨
```
Ascend: FAIL - Accuracy test failed  
```
This indicates compatibility issues with the Ascend backend.

### 🟡 Performance Concerns

#### 1. Small Shape Performance Regression
Some small shapes show significant performance degradation:
```
Shape [4, 16, 64, 4]:  0.627x (37% SLOWER)
Shape [16, 16, 8, 48]: 0.260x (74% SLOWER for bfloat16)
```

For small tensors, the Triton kernel overhead may outweigh benefits. Consider:
- Adding a threshold to fall back to PyTorch for small shapes
- Optimizing the kernel launch configuration for small batches

#### 2. BFloat16 Performance
The `[16, 16, 8, 48]` case with bfloat16 is particularly slow (0.260x). This should be investigated.

### 📊 Test Results Summary

| Backend | Status | Speedup |
|---------|--------|---------|
| Nvidia H20 | ✅ PASS | **1.90x** |
| Tianshu | ✅ PASS | **4.30x** |
| Muxi | ❌ FAIL | - |
| Ascend | ❌ FAIL | - |
| Hygon | ✅ PASS | - |

### ✅ What's Good

- Strong performance gains on supported backends (1.90x Nvidia, 4.30x Tianshu)
- Clean cuDNN interface wrapper design
- Comprehensive shape and dtype testing in benchmarks
- Good documentation of performance characteristics

### 📝 Recommendations

1. **Fix CI failure** - Investigate the `python-op` test failure
2. **Fix backend failures** - Debug Muxi and Ascend accuracy issues
3. **Consider small tensor optimization** - Add fallback for small shapes or optimize kernel launch
4. **Investigate bfloat16 regression** - The 0.260x case needs attention

### Verdict

**Changes Requested** ❌

This PR shows promising performance improvements but has critical CI and backend compatibility issues that must be resolved before merge.

**Action Items:**
1. ✅ Fix `python-op` CI failure
2. ✅ Fix Muxi backend accuracy issues
3. ✅ Fix Ascend backend accuracy issues
4. ✅ Investigate/fix small shape performance
5. ✅ Document bfloat16 performance characteristics

Looking forward to the updates! 🙏

File: src/flag_gems/__init__.py [hunk 1]
```diff
@@ -137,6 +137,7 @@ def torch_ge(v):
     ("avg_pool3d", avg_pool3d),
     ("avg_pool3d_backward", avg_pool3d_backward),
     ("baddbmm", baddbmm),
+    ("batch_norm_backward", batch_norm_backward_cudnn),
```
Comment (w1120029931-bit) [inline src/flag_gems/__init__.py:140]: why is *_cudnn？

**Comment by w1120029931-bit** [comment]
please resolve conflicts and unit-test error
